app/src/test/java/com/keylesspalace/tusky/StatusComparisonTest.kt

```diff
@@ -0,0 +1,216 @@
+        """.trimIndent()
+        assertNotEquals(createStatus(), createStatus(content = content))
+    }
+
+    @Test
+    fun `accounts with different notes in json - should be equal because notes are not relevant for timelines`() {
+        assertEquals(createStatus(note = "Test"), createStatus(note = "Test 123456"))
+    }
+
+    private val gson = GsonBuilder().registerTypeAdapter(
+        Spanned::class.java, SpannedTypeAdapter()
+    ).create()
+
+    @Test
+    fun `two equal status view data - should be equal`() {
+        val viewdata1 = StatusViewData.Concrete(
+            status = createStatus(),
+            isExpanded = false,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        val viewdata2 = StatusViewData.Concrete(
+            status = createStatus(),
+            isExpanded = false,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        assertEquals(viewdata1, viewdata2)
+    }
+
+    @Test
+    fun `status view data with different isExpanded - should not be equal`() {
+        val viewdata1 = StatusViewData.Concrete(
+            status = createStatus(),
+            isExpanded = true,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        val viewdata2 = StatusViewData.Concrete(
+            status = createStatus(),
+            isExpanded = false,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        assertNotEquals(viewdata1, viewdata2)
+    }
+
+    @Test
+    fun `status view data with different statuses- should not be equal`() {
+        val viewdata1 = StatusViewData.Concrete(
+            status = createStatus(content = "whatever"),
+            isExpanded = true,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        val viewdata2 = StatusViewData.Concrete(
+            status = createStatus(),
+            isExpanded = false,
+            isShowingContent = false,
+            isCollapsible = false,
+            isCollapsed = false
+        )
+        assertNotEquals(viewdata1, viewdata2)
+    }
+
+    private fun createStatus(
+        id: String = "123456",
+        content: String = """
+            \u003cp\u003e\u003cspan class=\"h-card\"\u003e\u003ca href=\"https://mastodon.social/@ConnyDuck\" class=\"u-url mention\" rel=\"nofollow noopener noreferrer\" target=\"_blank\"\u003e@\u003cspan\u003eConnyDuck@mastodon.social\u003c/span\u003e\u003c/a\u003e\u003c/span\u003e Hi\u003c/p\u003e
+        """.trimIndent(),
+        note: String = ""
+    ): Status {
+        val statusJson = """
```

Comment:
Did you make is JSON on purpose?

Reply:
Yes, so it gets created by Gson like in the real app.

Reply:
We could have a separate tests where we check that we parse correctly because in some cases statuses come not from JSON. It's not very important tho.